torch/fx/experimental/Partitioner.py

```diff
@@ -0,0 +1,151 @@
+           One partion only has to fit into one device
+           'backend type' is the type of that device
+        """
+        self.partition_id = partition_id
+        """
+           self.node_list contains graph nodes that will be deployed in this partition
+           List[[node_index, node_name, node_op]]
+           with node_ndex, it is easy to track if any node is missing from all partitions
+           node_name is to show what this node does
+           node_op is used to find placeholders for input node of the partition
+        """
+        self.node_list: List[[Tuple[int, str, str]]] = []
+
+    def add_node(self, node_index: int, node_name: str, node_op: str) -> None:
+        """append a new node into the partition"""
+        self.node_list.append((node_index, node_name, node_op))
+
+    def add_parent(self, partition: 'Partition') -> None:
+        self.parents.append(partition)
+
+    def add_child(self, partition: 'Partition') -> None:
+        self.children.append(partition)
+
+    def get_input_nodes(self) -> [int]:
+        """input nodes are coming from two places:
+        placeholder or output from its parents output
+        return a list of node indexes
+        """
+        input_nodes: List[Tuple[int, str, str]] = []
+        for node_index, node_name, node_op in self.node_list:
+            if node_op == 'placeholder':
+                input_nodes.append((node_index, node_name, node_op))
+        for partition_id in self.parents:
+            input_nodes.append(partition.get_output_node())
+        return input_nodes
+
+    def get_output_node(self) -> int:
+        """the output node of this partition should be the last node of self.node_list
+        """
+        assert len(self.node_list) != 0
+        return self.node_list[-1]
+
+    def __str__(self) -> str:
+        line: str = 'partition id: ' + str(self.partition_id) + '\n'
+        line += 'all nodes in this partition: '
+        for node_index, node_name, node_op in self.node_list:
+            line += '(node_' + str(node_index) + ':' + node_name +  ',' + node_op + ') '
+        line += '\n'
+        line += 'parent partitions:'
+        for parent in self.parents:
+            line += 'partition_' + str(parent.partitition_id) + ' '
+        line += '\n'
+        line += 'children partitions:'
+        for child in self.children:
+            line +=  str(child.partition_id) + ' '
+        line += '\n'
+        line += 'input nodes: '
+        input_nodes = self.get_input_nodes()
+        for node_index, node_name, node_op in input_nodes:
+            line += '(node_' + str(node_index) + ':' + node_name +  ',' + node_op + ') '
+        line += '\n'
+        output_node = self.get_output_node()
+        line += 'output node: ' + '(node_' + str(output_node[0]) + ':' + output_node[1] + ',' + output_node[2] + ')\n'
+        return line
+
+class Partitioner:
+    def __init__(self) -> None:
+        self.partitions: List[Partition] = []
+
+    """
```

Comment:
Per PEP8 https://www.python.org/dev/peps/pep-0257/ 
Docstrings should be the first thing below the declaration.